lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -1049,7 +1049,8 @@ using namespace Js;
 
     JavascriptArray* JavascriptOperators::GetOwnPropertyNames(Var instance, ScriptContext *scriptContext)
     {
-        RecyclableObject *object = VarTo<RecyclableObject>(ToObject(instance, scriptContext));
+        RecyclableObject *object = ToObject(instance, scriptContext);
+        AssertOrFailFast(VarIs<RecyclableObject>(object));
```

Comment:
nit: this doesn't _really_ make sense - imo, it's better structured as
```cpp
        AssertOrFailFast(VarIs<RecyclableObject>(instance));
        RecyclableObject *object = ToObject(instance, scriptContext); 
```
In the current form, you're saying "Is this RecyclableObject a RecyclableObject", which feels a bit off.

Reply:
Well, ToObject can allocate Number objects for number primitives, so this wouldn't actually be valid to check before. As you mentioned elsewhere, we can add a separate method for these tautological checks.

---
In reply to: [205901862](https://github.com/Microsoft/ChakraCore/pull/5456#discussion_r205901862) [](ancestors = 205901862)